### PR TITLE
groundwork for removing ci-operator

### PR DESCRIPTION
**Pull request by thekad**
```release-note
NONE
```
Trying to get rid of ci-operator, this is groundwork for removing ci-operator in the release repo

### CODE REVIEW

File: hack/tests/ci-prepare.sh [hunk 1]
```diff
@@ -30,6 +30,31 @@ stop_monitoring() {
...
 }
 
+cleanup() {
+    set +e
+
+    if [[ -n "$ARTIFACTS" ]]; then
+        exec &>"$ARTIFACTS/cleanup"
+    fi
+
+    if [[ -n "$MONITORING" ]]; then
```
Comment (asalkeld) [inline hack/tests/ci-prepare.sh]: do you need this check as stop_monitoring only does something if MON_PID is not empty

**Comment by asalkeld** [comment]
also s/Azure 186/groundwork for removing ci-operator in the PR headline so we know what it is about without having to go to the jira

File: hack/tests/ci-prepare.sh [hunk 2]
```diff
@@ -30,6 +30,31 @@ stop_monitoring() {
...
+      stop_monitoring
+    fi
+
+    make artifacts
+
+    if [[ -n "$T" ]]; then
```
Comment (jim-minter) [inline hack/tests/ci-prepare.sh]: the reason we hadn't condensed all the cleanup()s into one so far is that they're subtly different.  This PR doesn't yet handle things neatly enough to make me want to condense them.  Example: I don't want to rely on a variable called $T in a library cleanup().  Possibly renaming $T and making a wrapper function that cleans up $T then calls cleanup()?  Or still trapping the `rm -rf "$T"` in the other script?  Needs more experimentation.

File: hack/tests/e2e-create-prod.sh [hunk 3]
```diff
@@ -1,23 +1,8 @@
...
-
-    make artifacts
-
-    if [[ -n "$NO_DELETE" ]]; then
-        return
-    fi
-    make delete
-    az group delete -g "$RESOURCEGROUP" --yes --no-wait
-}
 trap cleanup EXIT
```
Comment (jim-minter) [inline hack/tests/e2e-create-prod.sh]: can the trap be moved to ci-prepare.sh?

File: hack/vmimage.sh [hunk 4]
```diff
@@ -17,7 +17,7 @@ cleanup() {
 }
 trap cleanup EXIT
```
Comment (jim-minter) [inline hack/vmimage.sh]: which cleanup actually gets called?  This one or the one in ci-prepare.sh?

File: hack/tests/e2e-create-prod.sh (hunk 3 above)
Comment (thekad) [reply hack/tests/e2e-create-prod.sh]: that would mean that `cleanup()` gets called at the end of every script, which may not be something you want to do, I rather keep it on each script to maintain some control

File: hack/tests/ci-prepare.sh (hunk 2 above)
Comment (thekad) [reply hack/tests/ci-prepare.sh]: Taking a look at all the cleanup functions they are mostly the same except for some of them having stop_monitoring, which as Angus points out is already idempotent: won't fail if monitoring wasn't started.
I do agree about T, I was just trying to maintain the status quo but that does look off to me, let me see how to best tackle that

File: hack/tests/e2e-create-prod.sh [hunk 5]
```diff
@@ -1,23 +1,8 @@
 #!/bin/bash -e
 
-cleanup() {
```
Comment (charlesakalugwu) [inline hack/tests/e2e-create-prod.sh]: it doesn't make sense to use the same cleanup() for the production e2e tests for reasons explained in https://github.com/openshift/openshift-azure/pull/1625

**Comment by charlesakalugwu** [comment]
flake #1628 
/retest

File: hack/tests/e2e-create-prod.sh (hunk 5 above)
Comment (thekad) [reply hack/tests/e2e-create-prod.sh]: better this way?

**Comment by mjudeikis** [comment]
so       much       space       here        :)
/lgtm

**Comment by charlesakalugwu** [comment]
/hold

Please can you improve the commit messages and/or squash this?

**Comment by mjudeikis** [comment]
/lgtm
/hold cancel
@charlesakalugwu thanks for this :) @thekad is out so doing this so we can clean PR queue :)

**Pull request merged**